File: problem_20.py
class FactorialDigitSum:

    def __init__(self, number):
        self.number = number

    def factorial(self):
        factorial = 1
        for x in range(1, self.number + 1):
            factorial *= x
        return factorial

    def digit_sum(self):
        digit_sum = 0
        for digit in list(str(self.factorial())):
            digit_sum += int(digit)
        return digit_sum

print(FactorialDigitSum(100).digit_sum())

File: problem_20.rb
#Calculate the sum of digits of the factorial of a given number.
class FactorialDigitSum

  private

  attr_accessor :number
  attr_writer :digit_sum

  def initialize(number)
    self.number = number
    self.digit_sum = sum_of_digits
  end

  def factorial
    (1..number).inject(:*)
  end

  def sum_of_digits
    factorial.to_s.split('').map { |n| n.to_i }.sum
  end

  public

  attr_reader :digit_sum

end

if __FILE__ == $PROGRAM_NAME
  puts FactorialDigitSum.new(100).digit_sum
end
